errors.go

```diff
@@ -37,6 +37,11 @@ var ErrShuttingDown = errors.New("kafka: message received by producer in process
 // ErrMessageTooLarge is returned when the next message to consume is larger than the configured Consumer.Fetch.Max
 var ErrMessageTooLarge = errors.New("kafka: message is larger than Consumer.Fetch.Max")
 
+var ErrInvalidArrayLength = PacketDecodingError{"invalid array length"}
+var ErrInvalidByteSliceLength = PacketDecodingError{"invalid byteslice length"}
+var ErrInvalidStringLength = PacketDecodingError{"invalid string length"}
+var ErrInvalidSubsetSize = PacketDecodingError{"invalid subset size"}
```

Comment:
I think it might make more sense to make these unexported and move them to the top of `real_decoder.go`, I can't think of a case where the caller would need to be able to distinguish between them.